### PR TITLE
Add seed to val shuffle for reproducibility

### PR DESCRIPTION
**Pull request by dfulu**
# Pull Request

## Description

With the streamed sample datamodule, we previously did not control the seeding in it. When training two different models which had the same global seed and same data config we could end up with completely different val samples. This fixes the issue and will make it easier to compare between models.

Additionally remove unused print_config param in example config

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pvnet/data/base_datamodule.py [hunk 1]
```diff
@@ -142,11 +145,16 @@ def setup(self, stage: str | None = None):
...
         if stage == "fit":
+
             # Prepare the train dataset
             self.train_dataset = self._get_streamed_samples_dataset(*self.train_period)
 
-            # Prepare and pre-shuffle the val dataset
+            # Prepare and pre-shuffle the val dataset and set seed for reproducibility
             val_dataset = self._get_streamed_samples_dataset(*self.val_period)
+
+            if self.seed is not None:
```
Comment (peterdudfield) [inline pvnet/data/base_datamodule.py:155]: worthing adding a log.info("Setting seed to X")?
